src/lora_region.erl

```diff
@@ -0,0 +1,1627 @@
+        ?_assertEqual(true, some_bit({0, 15}, [{0, 2}])),
+        ?_assertEqual(false, all_bit({0, 15}, [{0, 2}])),
+        ?_assertEqual(false, none_bit({0, 15}, [{0, 2}])),
+        ?_assertEqual(
+            [{link_adr_req, datar_to_dr('EU868', <<"SF12BW125">>), 14, 7, 0, 0}],
+            set_channels('EU868', {14, <<"SF12BW125">>, [{0, 2}]}, [])
+        ),
+        ?_assertEqual(
+            [
+                {link_adr_req, datar_to_dr('US915', <<"SF12BW500">>), 20, 0, 7, 0},
+                {link_adr_req, datar_to_dr('US915', <<"SF12BW500">>), 20, 255, 0, 0}
+            ],
+            set_channels('US915', {20, <<"SF12BW500">>, [{0, 7}]}, [])
+        ),
+        ?_assertEqual(
+            [
+                {link_adr_req, datar_to_dr('US915', <<"SF12BW500">>), 20, 2, 7, 0},
+                {link_adr_req, datar_to_dr('US915', <<"SF12BW500">>), 20, 65280, 0, 0}
+            ],
+            set_channels('US915', {20, <<"SF12BW500">>, [{8, 15}, {65, 65}]}, [])
+        )
+    ].
+
+mk_join_accept_cf_list_test_() ->
+    [
+        ?_assertEqual(
+            %% Active Channels 8-15
+            <<0, 255, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1>>,
+            mk_join_accept_cf_list('US915')
+        ),
+        ?_assertEqual(
+            %% Freqs 923.6, 923.8, 924.0, 924.2, 924.4
+            <<32, 238, 140, 240, 245, 140, 192, 253, 140, 144, 5, 141, 96, 13, 141, 0>>,
+            mk_join_accept_cf_list('AS923_1')
+        ),
+        ?_assertEqual(
+            %% Freqs 921.8, 922.0, 922.2, 922.4, 922.6
+            <<208, 167, 140, 160, 175, 140, 112, 183, 140, 64, 191, 140, 16, 199, 140, 0>>,
+            mk_join_accept_cf_list('AS923_2')
+        ),
+        ?_assertEqual(
+            %% Freqs 917.0, 917.2, 917.4, 917.6, 917.8
+            <<80, 236, 139, 32, 244, 139, 240, 251, 139, 192, 3, 140, 144, 11, 140, 0>>,
+            mk_join_accept_cf_list('AS923_3')
+        ),
+        ?_assertEqual(
+            %% Freqs 917.7, 917.9, 918.1, 918.3, 918.5
+            <<168, 7, 140, 120, 15, 140, 72, 23, 140, 24, 31, 140, 232, 38, 140, 0>>,
+            mk_join_accept_cf_list('AS923_4')
+        ),
+        ?_assertEqual(
+            %% Freqs 867.1, 867.3, 867.5, 867.7, 867.9
+            <<24, 79, 132, 232, 86, 132, 184, 94, 132, 136, 102, 132, 88, 110, 132, 0>>,
+            mk_join_accept_cf_list('EU868')
+        )
+    ].
+
+%%--------------------------------------------------------------------
+%% Private Utilities
+%%--------------------------------------------------------------------
+
+some_bit(MinMax, Chans) ->
+    lists:any(
+        fun(Tuple) -> match_part(MinMax, Tuple) end,
+        Chans
+    ).
+
+none_bit(MinMax, Chans) ->
+    lists:all(
+        fun(Tuple) -> not match_part(MinMax, Tuple) end,
+        Chans
+    ).
+
+match_part(MinMax, {A, B}) when B < A ->
+    match_part(MinMax, {B, A});
+match_part({Min, Max}, {A, B}) ->
+    (A =< Max) and (B >= Min).
+
+-endif.
+%% end of file
```

Comment:
There are no tests exercising Rx Delay.

Some may be extracted from router repo: `test/router_SUITE.erl` such as:
-  rx_delay_join_test/1,
-  rx_delay_downlink_default_test/1,
-  rx_delay_change_ignored_by_device_downlink_test/1,
-  rx_delay_accepted_by_device_downlink_test/1,
-  rx_delay_continue_session_test/1,
-  rx_delay_change_during_session_test/1

Reply:
lora_region is now renamed test_region.
Also test_region is now protected with EUNIIT ifdef